libs/vertexai/langchain_google_vertexai/vectorstores/document_storage.py

```diff
@@ -39,16 +43,26 @@ def mset(self, key_value_pairs: Sequence[Tuple[str, Document]]) -> None:
         Args:
             key_value_pairs (Sequence[Tuple[K, V]]): A sequence of key-value pairs.
         """
-        for key, value in key_value_pairs:
-            self._set_one(key, value)
+        with TemporaryDirectory() as tmp_folder:
+            for key, value in key_value_pairs:
+                with open(Path(tmp_folder) / key, "w") as f:
+                    json.dump(value.dict(), f)
+
+            transfer_manager.upload_many_from_filenames(
+                self._bucket,
+                [item[0] for item in key_value_pairs],
+                blob_name_prefix=f"{self._prefix}/",
+                source_directory=tmp_folder,
+                upload_kwargs={"retry": DEFAULT_RETRY},
+            )
```

Comment:
Looks pretty good! The only thing I'm not quite sure about is the use of the disk to stage the files, have you tried `upload_many` instead?
https://cloud.google.com/python/docs/reference/storage/latest/google.cloud.storage.transfer_manager

Reply:
I've looked at it but I don't think it's the right choice because often uploading documents includes large amount of files which will not fit in memory